dashboard/app/controllers/data_docs_controller.rb

```diff
@@ -62,4 +56,9 @@ def data_doc_params
       :content
     )
   end
+
+  def set_data_doc
+    @data_doc = DataDoc.find_by(key: params[:key])
+    return render :not_found unless @data_doc
```

Comment:
not to keep on this but I've struggled with rendering not found in the private method. Does this work correctly? If so, looks great!

Reply:
I re-tested locally and was able to go through all the routes of creating, editing, and submitting data docs so I think it still works correctly! I'm waiting to make sure all drone tests still pass just in case but I think it should be good

Reply:
Hi @TurnerRiley , what happens when you do one of these actions (edit or update) with a key that does not exist? that's what needs to be tested to evaluate Bethany's concern.

Reply:
just closing the loop, it looks like this is giving a 404 as expected: http://localhost-studio.code.org:3000/data_docs/foo/edit